pkg/kcp/nuke/resourceStatusDeleted.go

```diff
@@ -4,14 +4,18 @@ import (
 	"context"
 	cloudcontrolv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-control/v1beta1"
 	"github.com/kyma-project/cloud-manager/pkg/composed"
+	"github.com/kyma-project/cloud-manager/pkg/kcp/nuke/types"
 )
 
 func resourceStatusDeleted(ctx context.Context, st composed.State) (error, context.Context) {
-	state := st.(*State)
+	state := st.(*types.State)
 
 	changed := false
 
 	for _, sk := range state.ObjAsNuke().Status.Resources {
+		if sk.ResourceType != "" && sk.ResourceType != cloudcontrolv1beta1.KcpManagedResource {
```

Comment:
Maybe it would be best to add a getter method on the type that will `if empty return KCP` to avoid these double checks